### ISSUE TITLE
Word add-in manifest validates correctly, but fails on centralized deployment

### ISSUE THREAD

**Issue by JorisZzz** [OfficeDev/generator-office#593]
I am working on a Word taskpane add-in (but so far I am only using ribbon elements through Action:ExecuteFunction).
When I try to upload my manifest.xml via centralized deployment, it fails with the message "Manifest file validation has failed. {GUID}"

**npm run validate** returns "The manifest is valid"
The add-in runs correctly on my local machine.
I have executed **npm run build** and I am trying to upload the resulting manifest.prod.xml. 
I have uploaded the entire contents of the dist folder to a webserver and checked it's accessibility.

I am running Node v14.15.1, I have done **npm update** and **npm i office-addin-manifest -D** (now on 1.5.12)

When I create a new Yo Office solution, don't change anything, build it, and upload the manifest through centralized deployment, everything works fine.

Since the error message is non specific I am hoping someone can point me to a solution, or tell me how to go about troubleshooting this issue?

**Comment by JorisZzz** [comment]
So.. after going through my manifest line by line, it turns out this is the culprit: 
<InsertAfter>TabHome</InsertAfter>

Somehow the validator did not find a problem with this entry, but centralized deployment does not accept it.

If anyone has a working example for <InsertAfter> or <InsertBefore> I would appreciate that.

Thanks.

**Comment by arttarawork** [comment]
@JorisZzz Thank you for bringing this issue to our attention! I'll try my best to replicate the issue and give you a solution. Is the repo to your taskpane add-in available publicly?

**Comment by JorisZzz** [comment]
> @JorisZzz And running `office-addin-manifest validate manifest.prod.xml` succeeds?

Yes, it does.

**Comment by JorisZzz** [comment]
> @JorisZzz Thank you for bringing this issue to our attention! I'll try my best to replicate the issue and give you a solution. Is the repo to your taskpane add-in available publicly?

No, it's not. But here is the manifest (with altered links). The offending element is commented out on line# 243
[manifest.prod.zip](https://github.com/OfficeDev/generator-office/files/5675777/manifest.prod.zip)

**Comment by ralphtheninja** [comment]
> Somehow the validator did not find a problem with this entry, but centralized deployment does not accept it.

It would be nice if the centralized deployment could use the same code, e.g. `office-addin-manifest` so the logic is same everywhere. Maybe it is, but just dated? *shrug*

**Comment by DChefHarshil** [comment]
@JorisZzz Thanks for raising this. I am facing the same issue. I am developing a word-addin. I am able to validate the add-in manifest using npm run validate but when I try upload the file to word online, it throws an error saying the manifest is not valid.

**Comment by millerds** [comment]
The command to validate the manifest is a service call that check for store submission validity.  I believe that centralized deployment has some other restrictions that are not the same as the store.  This is something that could be added to the validation endpoint.
